src/agent/solana/exporter.rs

```diff
@@ -501,6 +494,31 @@ impl Exporter {
                     false
                 }
             })
+            .filter(|(id, info)| {
+                // Filtering out prices that are being updated too frequently according to publisher_permission.publish_interval
+                let last_info = self.last_published_state.get(id);
+                if last_info.is_none() {
+                    // No prior data found, letting the price through
+                    return true;
+                }
+                let last_info = last_info.unwrap();
```

Comment:
`match last_info { ... }` is cleaner here.

Reply:
We have 3 `Options` here that we have to unwrap. I initially did the `match` and `let Some(...)` but it looked really ungly and unreadable. I think this way atleast it avoids 3 layers of nested matches.

Reply:
You can make it:
```rust
let last_info = match self.last_published_state.get(id) {
  Some(info) => info,
  None => {
    return true,
  }
}
```

Reply:
Perfect. Thanks.